code/Control/UserDefinedFormController.php

```diff
@@ -268,6 +274,12 @@ public function process($data, $form)
                             return;
                         }
 
+                        // generate image thumbnail to show in asset-admin
+                        // you can run userforms without asset-admin, so need to ensure asset-admin is installed
+                        if (class_exists(AssetAdmin::class) && $file->getIsImage()) {
```

Comment:
You don't need to validate that $file is an Image. `generateThumbnails` does that already. 

If we ever implement a feature allows thumbnail generation for non-Image files, this condition will get in the way.

```suggestion
                        if (class_exists(AssetAdmin::class)) {
```

Reply:
Updated